drivers/quota/projectquota_supported.go

```diff
@@ -370,6 +370,20 @@ func setProjectID(targetPath string, projectID uint32) error {
 		return fmt.Errorf("failed to set projid for %s: %w", targetPath, errno)
 	}
 
+	// While we are setting the PROJINHERIT flag, it only applies to files
+	// and directories created after the fact.
+	// Any pre-existing subdirectories will still need to have the project
+	// ID set manually, so recurse onto them.
+	if dents, err := os.ReadDir(targetPath); err == nil {
+		for _, dent := range dents {
+			if dent.IsDir() {
+				if err := setProjectID(filepath.Join(targetPath, dent.Name()), projectID); err != nil {
+					return fmt.Errorf("recursively setting project ID on subdirectories of %s: %w", targetPath, err)
+				}
```

Comment:
I am not sure if it matters for our use case but if we walk a dir and then try to open the path it is possible that it has been removed between the listing and the open call.
As such I think this API needs to ignore ErrNotExist (though as this uses cgo I have not checked if matching ErrNotExist will actually match the right error in that case, I think it does but better double check)